src/tests/ftest/server/metadata.py

```diff
@@ -380,27 +374,23 @@ def test_metadata_server_restart(self):
                     "F", "/run/ior/ior{}flags/".format(operation))
 
                 # Define the job manager for the IOR command
-                manager = Orterun(ior_cmd)
-                env = ior_cmd.get_default_env(str(manager))
-                manager.assign_hosts(self.hostlist_clients, self.workdir, None)
-                manager.assign_processes(processes)
-                manager.assign_environment(env)
+                self.ior_managers.append(Orterun(ior_cmd))
+                self.ior_managers[-1].tmpdir_base.update(self.test_dir, "orterun.tmpdir_base")
+                env = ior_cmd.get_default_env(str(self.ior_managers[-1]))
+                self.ior_managers[-1].assign_hosts(self.hostlist_clients, self.workdir, None)
+                self.ior_managers[-1].assign_processes(processes)
+                self.ior_managers[-1].assign_environment(env)
+                self.ior_managers[-1].verbose = False
 
                 # Add a thread for these IOR arguments
-                threads.append(
-                    threading.Thread(
-                        target=ior_runner_thread,
-                        kwargs={
-                            "manager": manager,
-                            "uuids": list_of_uuid_lists[index],
-                            "results": self.out_queue}))
-
+                thread_manager.add(manager=self.ior_managers[-1], uuids=list_of_uuid_lists[index])
                 self.log.info(
                     "Created %s thread %s with container uuids %s", operation,
                     index, list_of_uuid_lists[index])
 
             # Launch the IOR threads
-            if self.thread_control(threads, operation) == "FAIL":
+            self.log.info("Launching %d IOR %s threads", thread_manager.qty, operation)
+            if thread_manager.check_run():
                 self.d_log.error("IOR {} Thread FAIL".format(operation))
```

Comment:
this error will not indicate how many threads failed (one, a subset, or all) so the message may be a little confusing "Thread FAIL" as if it is singular. Maybe a small issue not worth making a substantial change for, but would it make sense for check_run() to return a number (0 = no failures all threads succeeded, and >0 indicating some failures, and the number of threads that failed)?

Reply:
That information is provided in the log output by the `ThreadManager.check()` method called via `thread_manager.check_run()`, but I can make the change you've suggested to make the error summary clearer.